src/cnxml/deserialize.js

```diff
@@ -78,6 +83,27 @@ const MARK = {
  */
 const DEFAULT = {
     deserialize(el, next) {
+        // We do not support fully those elements yet so for now they are 
+        // transformed into normal text so magicians can edit them in 
+        // source mode. 
+
+        const handleInSourceMode = [
+            'math', 'rule', 'statement', 'proof', 'equation', 'foreign'
+        ]
+
+        if (el.nodeName !== '#text' && handleInSourceMode.includes(el.localName.toLowerCase())) {
+            const parentsForInlines = ['para', 'caption']
+            let data = {
+                object: 'block',
+                type: 'source_element',
+                nodes: [Text.create(el.outerHTML)],
```

Comment:
`el.outerHTML` will serialize the element as HTML, which is different from (and incompatible with) XML.

Reply:
From what I've tested it's giving exactly the same output of element which was in the module content.

Reply:
One example: HTML has self-closing tags, XML doesn't. Using `<br>` as an example:

```js
root = document.createElement('root')
br = document.createElement('br')
root.appendChild(br)
new XMLSerializer().serializeToString(root)
// => "<root xmlns=\"http://www.w3.org/1999/xhtml\"><br /></root>"
root.outerHTML
// => "<root><br></root>"
new DOMParser().parseFromString(root.outerHTML, 'application/xml')
// => XML Parsing Error: mismatched tag. Expected: </br>.
```

Reply:
But we are deserializing already parsed, valid xml?

Reply:
`el` is not XML, it's a DOM element deserialized from XML. DOM has no knowledge of whether it came from HTML or XML, and it doesn't matter. `el.outerHTML` is a computed property which, whenever accessed, will serialize this particular DOM element to HTML or XML, depending on the type of document it came from (had to check reference to verify that, I was convinced it always serialized to HTML). Nonetheless it will always be safer to explicitly serialize to XML.